docs: extract section about target cluster configuration (#2792)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/configuring/con-configuring-vc-target.adoc

Examples below illustrate the various strategies which can be used for selecting a bootstrap server from the list of addresses configured.

When multiple bootstrap servers are configured with no bootstrap server selection strategy explicitly configured, the bootstrap server is selected using the `round-robin` strategy.

.Example using default strategy
[source,yaml]
----
#...
virtualClusters:
    - name: my-cluster-proxy
      # ...
      targetCluster:
        bootstrapServers: my-cluster-kafka-bootstrap-0.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-1.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-2.kafka.svc.cluster.local:9093
        tls: {}
#...
----

The below example illustrates explicitly configuring a bootstrap server selection strategy.

.Example configuring a bootstrap server selection strategy
[source,yaml]
----
#...
virtualClusters:
  - name: my-cluster-proxy
    #...
    targetCluster:
      bootstrapServers: my-cluster-kafka-bootstrap-0.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-1.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-2.kafka.svc.cluster.local:9093
      bootstrapServerSelection:

>         strategy: "round-robin" # <1>

Reviewer: end the example with `#...`

----

<1> Round-robin strategy for bootstrap server selection.

> Other supported strategy is `random`.

Reviewer: It feels a bit funky to enumerate the options here, I think we should do it higher up using bullets with a sentence each to distinguish round robin and random. (though I think having this note here to highlight the relevant setting is fine).

////
  // Copyright Kroxylicious Authors.
  //
  // Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
////
:_mod-docs-content-type: CONCEPT

[id='con-configuring-vc-target-{context}']
= Configuring a target cluster

[role="_abstract"]

> The target cluster specifies the physical upstream Kafka cluster which is proxied by the `my-cluster-proxy` virtual cluster.

Reviewer 2: ```suggestion
The target cluster defines the physical upstream Kafka cluster that is proxied by the `my-cluster-proxy` virtual cluster.
```

The target cluster configuration has 3 properties.

> 1. `bootstrapServers`: Comma-separated list of bootstrap address(es) for the Kafka cluster being proxied.   If the cluster exposes a bootstrap endpoint (as is the case with some cloud-based Kafka services or clusters managed by Strimzi), use that address.  If the cluster does not have a specific bootstrap endpoint (as is the case with a traditional kafka deployment) list the address of several broker nodes. This helps spread the bootstrapping load and also provides better reliability.

Reviewer 2: ```suggestion
1. `bootstrapServers`: Comma-separated list of bootstrap addresses for the Kafka cluster being proxied.   If the cluster exposes a bootstrap endpoint (as is the case with some cloud-based Kafka services or clusters managed by Strimzi), use that address.  If the cluster does not have a specific bootstrap endpoint (as is the case with a standard Kafka deployment) list the address of several broker nodes. This helps spread the bootstrapping load and improves reliability.
```

////
  // Copyright Kroxylicious Authors.
  //
  // Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
////
:_mod-docs-content-type: CONCEPT

[id='con-configuring-vc-target-{context}']

> = Configuring target cluster

Reviewer 3: ```suggestion
= Configuring a target cluster
```

[role="_abstract"]
The target cluster specifies the physical upstream Kafka cluster which is proxied by the `my-cluster-proxy` virtual cluster.

The target cluster configuration has 3 properties.

1. `bootstrapServers`: The bootstrap addresses of the physical Kafka cluster being proxied. You can specify multiple addresses by providing a comma separated list. If you want to bootstrap to individual Kafka brokers, then having multiple addresses helps spread the bootstrapping load and also provides better reliability.
+

> NOTE: If the target cluster is a *Strimzi* managed Kafka cluster, you can specify the address of the *bootstrap service* which takes care of distributing the requests.

Reviewer 3: I think this needs to be generalised as it is not just Strimzi that behaves in that way. How about:

```suggestion
NOTE: If the target cluster exposes a bootstrap endpoint (as is the case with some cloud-based Kafka services or clusters managed by Strimzi), specify its bootstrap endpoint in preference to enumerating the individual broker addresses.
```

This property configures the strategy used for selecting the bootstrap server. +
The following strategies are supported:
.. `round-robin`: This strategy selects a server from the configured list of servers in a round-robin manner starting from the first server looping back after the last server is selected. This is default strategy used when no strategy is explicitly configured.
.. `random`: This strategy selects a random server from the configured list of servers.
3. `tls`: TLS configuration for connecting to the target Kafka cluster.
See xref:con-configuring-vc-target-tls-{context}[Securing connections to target clusters] for more details.

Examples below illustrate the various strategies which can be used for selecting a bootstrap server from the list of addresses configured.

When multiple bootstrap servers are configured with no bootstrap server selection strategy explicitly configured, the bootstrap server is selected using the `round-robin` strategy.

.Example using default strategy
[source,yaml]
----
#...
virtualClusters:
    - name: my-cluster-proxy
      # ...
      targetCluster:

>         bootstrapServers: my-cluster-kafka-bootstrap-0.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-1.kafka.svc.cluster.local:9093, my-cluster-kafka-bootstrap-2.kafka.svc.cluster.local:9093

Reviewer 3: Not sure we actually support spaces in the list.

```suggestion
        bootstrapServers: my-cluster-kafka-bootstrap-0.kafka.svc.cluster.local:9093,my-cluster-kafka-bootstrap-1.kafka.svc.cluster.local:9093,my-cluster-kafka-bootstrap-2.kafka.svc.cluster.local:9093
```
